src/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -4152,46 +4152,14 @@ public static boolean isArrayObject(Object obj)
     static void checkDeprecated(Context cx, String name) {
         int version = cx.getLanguageVersion();
         if (version >= Context.VERSION_1_4 || version == Context.VERSION_DEFAULT) {
-            String msg = getMessage1("msg.deprec.ctor", name);
+            String msg = getMessage("msg.deprec.ctor", name);
             if (version == Context.VERSION_DEFAULT)
                 Context.reportWarning(msg);
             else
                 throw Context.reportRuntimeError(msg);
         }
     }
 
-    public static String getMessage0(String messageId)
```

Comment:
Unfortunately, we have no way of knowing how much third-party code built on Rhino uses this and the other "public static" methods. I'm totally sure that it's more than zero. So while I think it's great to modernize the codebase and update all the places that use these functions, I don't think that we should delete them. I would prefer if you instead marked them with a deprecated annotation.

Reply:
Thanks. I made getMessageById and deprecated old method.